.github/workflows/main.yml

```diff
@@ -0,0 +1,87 @@
+name: Veracruz-CI
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push or pull request events but only for the main branch
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+env:
+  WASMTIME_VERSION: v9.0.1
+jobs:
+  vod-deployment:
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/veracruz-project/veracruz/ci@sha256:38add618b42f59a89fdd4c46831dcf9c9e73b159bc74c87251c8263a2ec6238a
+      volumes:
+        - ${{ github.workspace }}:/work/video-object-detection
+    steps:
+      - name: Check out the VOD repository
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+          set-safe-directory: true
+      - name: Build
+        run: |
+          # grab every bash code block for this step, remove line continuation,
+          # and only keep lines that start with '$' (of course removing that '$'
+          # in the process)
+          sed -n '/```.*vod-ci-build/,/```/{/```/d; p}' README.md \
```

Comment:
This incantation is now repeated ten times across our repos! Would it be better to have a Perl script in each of the two repos that we invoke from the `main.yml`? (I don't think we should delay this PR because of that, though.)

Reply:
Looks like we've finally found a good use case for composite actions. Yeah let's do that in another PR